pvnet/models/base_model.py

```diff
@@ -141,6 +142,65 @@ def minimize_data_config(input_path, output_path, model):
         yaml.dump(config, outfile, default_flow_style=False)
 
 
+def download_hf_hub_with_retries(
+    repo_id,
+    filename,
+    revision,
+    cache_dir,
+    force_download,
+    proxies,
+    resume_download,
+    token,
+    local_files_only,
+    max_retries=5,
+    wait_time=60,
+):
+    """
+    Tries to download a file from HuggingFace up to max_retries times.
+
+    Args:
+        repo_id (str): HuggingFace repo ID
+        filename (str): Name of the file to download
+        revision (str): Specific model revision
+        cache_dir (str): Cache directory
+        force_download (bool): Whether to force a new download
+        proxies (dict): Proxy settings
+        resume_download (bool): Resume interrupted downloads
+        token (str): HuggingFace auth token
+        local_files_only (bool): Use local files only
+        max_retries (int): Maximum number of retry attempts
+        wait_time (int): Wait time (in seconds) before retrying
+
+    Returns:
+        str: The local file path of the downloaded file
+    """
+    for attempt in range(1, max_retries + 1):
+        try:
+            return hf_hub_download(
+                repo_id=repo_id,
+                filename=filename,
+                revision=revision,
+                cache_dir=cache_dir,
+                force_download=force_download,
+                proxies=proxies,
+                resume_download=resume_download,
+                token=token,
+                local_files_only=local_files_only,
+            )
+        except Exception as e:
+            logging.warning(
+                (
+                    f"Attempt {attempt}/{max_retries} failed to download {filename}. "
+                    f"Retrying in {wait_time} seconds..."
+                )
+            )
+            if attempt == max_retries:
```

Comment:
could this if statement go above the `logging.warning`, then the logging will read slightly more clear if it does fail